DNN Platform/Tests/DotNetNuke.Tests.Integration/Tests/JwtAuthTest.cs

```diff
@@ -0,0 +1,344 @@
+#endif
+
+        private const string LoginQuery = "/DesktopModules/JwtAuth/API/mobile/login";
+        public const string ExtendTokenQuery = "/DesktopModules/JwtAuth/API/mobile/extendtoken";
+        private const string TestGetQuery = "/DesktopModules/JwtAuth/API/mobile/testget";
+        private const string TestPostQuery = "/DesktopModules/JwtAuth/API/mobile/testpost";
+
+        public JwtAuthTest()
+        {
+            var url = ConfigurationManager.AppSettings["siteUrl"];
+            var siteUri = new Uri(url);
+            _httpClient = new HttpClient { BaseAddress = siteUri, Timeout = _timeout };
+            _hostName = ConfigurationManager.AppSettings["hostUsername"];
+            _hostPass = ConfigurationManager.AppSettings["hostPassword"];
+        }
+
+        #endregion
+
+        #region tests
+
+        [Test]
+        public void InvalidUserLoginShouldFail()
+        {
+            var credentials = new { u = _hostName, p = _hostPass + "." };
+            var result = _httpClient.PostAsJsonAsync(LoginQuery, credentials).Result;
+            var content = result.Content.ReadAsStringAsync().Result;
+            ShowInfo(@"content => " + content);
+            Assert.AreEqual(HttpStatusCode.Unauthorized, result.StatusCode);
+        }
+
+        [Test]
+        public void ValidUserLoginShouldPass()
+        {
+            var token = GetAuthorizationTokenFor(_hostName, _hostPass);
+            Assert.IsNotNull(token.AccessToken);
+            Assert.IsNotNull(token.DisplayName);
+            Assert.IsNotNull(token.RenewalToken);
+
+            var parts = token.AccessToken.Split('.');
+            var decoded = DecodeBase64(parts[1]);
+            dynamic claims = JsonConvert.DeserializeObject(decoded);
+            long claimExpiry = claims.exp;
+            var expiryInToken = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc).AddSeconds(claimExpiry);
+            Assert.Less(DateTime.UtcNow, expiryInToken);
+            Assert.LessOrEqual(expiryInToken, DateTime.UtcNow.AddHours(1));
+        }
+
+        [Test]
+        public void RequestWithoutTokenShouldFail()
+        {
+            var result2 = _httpClient.GetAsync(TestGetQuery).Result;
+            var content2 = result2.Content.ReadAsStringAsync().Result;
+            ShowInfo(@"content2 => " + content2);
+            Assert.AreEqual(HttpStatusCode.Unauthorized, result2.StatusCode);
+        }
+
+        [Test]
+        public void ValidatedUserGetRequestShouldPass()
+        {
+            var token = GetAuthorizationTokenFor(_hostName, _hostPass);
+            SetAuthHeaderToken(token.AccessToken);
+            var result = _httpClient.GetAsync(TestGetQuery).Result;
+            var content = result.Content.ReadAsStringAsync().Result;
+            ShowInfo(@"content2 => " + content);
+            Assert.AreEqual(HttpStatusCode.OK, result.StatusCode);
+            Assert.Less(0, content.IndexOf("You are authenticated through JWT", StringComparison.Ordinal));
```

Comment:
:cry: what is this world coming to when NUnit doesn't have an assertion which includes a `StringComparison`?